Validation/RecoParticleFlow/bin/include/HistoData.h

```diff
@@ -90,7 +90,7 @@ class HistoData {
 
   // Misc Functions
   void drawResult(TH1 *Summary, bool Vertical = true, bool SetLabels = false);
-  void clear() { memset(static_cast<void *>(this),0,sizeof(*this)); }
+  void clear(){ newHisto->Clear(); refHisto->Clear(); };
```

Comment:
Is the intent of `clear()` to only empty out the histograms or to also reset the other member data to `0`?

Reply:
It is only used here https://github.com/gartung/cmssw/blob/cbb473572c44b455b5edbd5d64cf14394e111a9a/Validation/RecoParticleFlow/bin/include/PlotCompareUtility.h#L97

Reply:
The line you are pointing to clears a `std::vector<HistoData>` not the `HistoData` itself. I would  guess that if someone actually called `HistoData::clear` any subsequent calls to that object would cause a segmentation fault.

Reply:
DXR claims this clear function is never called. Should I just remove it?

Reply:
Looking further at the code, I think your new version of `clear` is probably the intent since all the other member variables are set via a `set` function so it does not seem to make sense to rest them to `0` or empty.

Reply:
The hell is.... was.... `void clear() { memset(static_cast<void *>(this),0,sizeof(*this)); }`
This definitely goes into a collection of wonderful code snippets.